### DISCUSSION TITLE
Branch 6.4

### DISCUSSION THREAD

**Discussion by xillibit** [Kunena/Kunena-Forum#9686] [General]
@Ruud68 : for info i have just created the branch 6.4, this branch drops the support of Joomla! 4.4.x and it allow to be installed on Joomla! 5.2. The requirements has been updated on this branch to follow ones defined for Joomla! 5.x

When the developpement of Php 8.4 will be more advanced this branch will support too this version

**Comment by Ruud68** [comment]
Thanks @xillibit
Think it is good to discuss some things as to what we would like to add / refactor for this branch.
e. using form / views for e.g. category / smiley / etc. implementing the Core Joomla classes and building on top of that (like we did for the list views categories, users, etc.).
Also remove the BS modals in favor of the new dialog popups that J5 provides (we now have both implemented so this is basically a cleanup)

**Comment by xillibit** [comment]
> e. using form / views for e.g. category / smiley / etc. implementing the Core Joomla classes and building on top of that (like we did for the list views categories, users, etc.).

you mean in backend ?

I had in mind to improve the install part to put back something which has been removed around K5.1 or K5.2 since a change in Joomla! itself. Before after the unpack of kunena package, there was a javascript which make the update of kunena tables since the Joomla! change this has been done when the user open for the first time the Kunena dashboard. Some users forgot to do that...

**Reply by Ruud68** [reply]
Yes, backend. I think that while in version 6 focus should be on backend stuff, but that is my opinion:)

Updating of tables I do myself in the update script which act on the update function. No JS needed. Need to investigate how it works now to see if we can easily do it via update script.

Maybe we open a discussion topic om the forum asking people what they would like to have changes / added and take it from there.
I think that can also 'spark' the community so we get more testers and possibly developers

**Reply by xillibit** [reply]
During update they can be mysql queries and/or php script to make specific changes

There are too deprecated thing in Joomla! 5.x used in Kunena will be removed in Joomla! 6.0, like some Kunena classes extend the CMSObject class and that's will need to be changed at some point

**Reply by Ruud68** [reply]
Difficulty for me is that when years back in the Joomla dev community there was discussion on following the HMVC architecture. I recall Kunena jumping in and adapting everything to HMVC at which point Joomla desided not to do that but started following the MVC architecture.

For me that complicates things as I never got my head around HMVC (not being used in anything else I'm working with).

That makes it a steep learning curve for contributors (at least that is what I experience). So for me it would make sense if we could refactor more into (Joomla) MVC extending where possible Joomla classes and using traits for specific Kunena functions that should be used generic over multiple classes.

I am also active on other forums and when I see what functionality they have / how it looks (discourse, flarum, etc.), then I'm afraid of Kunena lagging behind in functionality / extensibility.

**Comment by Ruud68** [comment]
Just as another thought. I see two directions in software development: do more with core (basically what the big software vendors are doing: build in more functionality into the core), or do less with core and make core extensible.
I think Kunena would benefit from the do less approach: back to what is basic for a forum and do that in an awesome way. Open Kunena via API's / hooks so that software developers can jump in and easily add (and maintain) functionality.

As an example: Nextcloud. They offer a kick-ass platform and have a very vibrant developers community build that offers apps that add functionality to the core functionlity.

Kunena should IMO become more a platform with basic forum functionality and things like twitter / bluesky / masotdo / etc. integration should come from 'external' devs.

Just a thought :)

I think this is worth a discussion

**Comment by AllonisDavid** [comment]
> @Ruud68 : for info i have just created the branch 6.4, this branch drops the support of Joomla! 4.4.x and it allow to be installed on Joomla! 5.2. The requirements has been updated on this branch to follow ones defined for Joomla! 5.x
> 
> When the developpement of Php 8.4 will be more advanced this branch will support too this version

Currently on Joomla 4.4.14....doing the research to upgrade it.
System / Integrations / Extension / Upgrades is suggesting we upgrade to K 6.4.9  But if I am reading your note - that is not compatible with 4.4.x...
Trying to research this - the documentation is inconsistent on what version of K goes to which version of Joomla.

If you advise that we should upgrade K on current 4.4.14 joomla....and then we upgrade Joomla to J5.x - all should be good again?
Thanx for the assist!!!

**Reply by xillibit** [reply]
Hello,

On which Kunena version are-you actually ? Only Kunena 6.3 support Joomla! 4.4.x, Joomla! 5.0.x, 5.1.x and 5.2.x

**Reply by AllonisDavid** [reply]
K is currently 6.3.10.  and Joomla 4.4.14
and K is suggesting to upgrade to 6.4.9 (which seems wrong based on documentation).
and we are attempting to upgrade Joomla to 5.x

K seems to be working fine with aurelia template 6.3.10 and NTSKay 6.2.7 on today's server config.

thanx!!!

**Reply by Ruud68** [reply]
Hi, As far as I know, Joomla handles this. When you are not prompted with a new version for kunena (assuming you didn't disable the update site, then you are on the latest working version for that Joomla version. When you now upgrade Joomla to version 5, the preupdate checker will tell you if you can or cannot due to incompatible extension versions: these should then first be upgraded. normally the latest Kunena version supports also the next Joomla version (otherwise you could never upgrade Joomla). When you upgrade to Joomla 5, then you will be prompetd that there is a newer versiion for kunena available and you can install that.
Always make backups between these bigger (extensio) updates

**Reply by AllonisDavid** [reply]
The Joomla Upgrader IS suggesting that we upgrade K before upgrading.  But that looks like it will break it working on J 4.4.14 (before we upgrade J).  Yes we are backing up.

**Reply by xillibit** [reply]
Kunena doesn't know which version of Joomla! you are using when it suggest an update. So upgrade first to Joomla! 5.2 then you can update Kunena to 6.4.9 and then update Joomla! to 5.4.x

**Reply by AllonisDavid** [reply]
Ok, but the Pre-Update Checker advises to update K First before upgrading J.  Confusing...
I assume it's unfortunate that K doesn't know what v of J it is installed on before suggesting an update....can't that break things badly?
BTW, thank you both for your tips!

**Reply by xillibit** [reply]
In the update files there is the versions of Joomla! supported by Kunena i don't know why the Joomla! updater check that before forcing to update some extensions. I will remove in Kunena files for K6.3 users the suggestion to update to K6.4.9, just wait a couple of hours. Then you will need to update manually to Joomla! 5.2.x, then update Kunena to 6.4.9 and then let Joomla! update itself to 5.4.x

**Reply by xillibit** [reply]
I have disabled the suggested update to K6.4.9 when users are on K6.3.10

So now you can make a backup and update manualy to Joomla! 5.2, and then follow what i had described in my message above

**Reply by AllonisDavid** [reply]
Thanx xillibit, will do and post results...thanx mucho!

**Reply by AllonisDavid** [reply]
Joomla is still suggesting that I upgrade to 6.4 in the normal updater
The Pre-update check no longer "forces" a Kunena upgrade.
But it does show as "Joomla cannot detect the extension's compatibility with the target version of Joomla."
for Kunena - CB AntiSpam and
pkg_kunena- System - Kunena Forum  Potential Upgrade Issue.

FYI

**Reply by AllonisDavid** [reply]
So, in researching doing a "manual" install of 5.2....seems most are saying doing a FTP upgrade is really frowned upon...is there a better way you suggest?

**Reply by xillibit** [reply]
It's the only way, download the package of Joomla! 5.2.x, unzip it in a folder in your computer and then transfer with FTP client all the content of this folder.

**Reply by AllonisDavid** [reply]
Thank you for the confirmation.  I'll give it a go.

**Reply by AllonisDavid** [reply]
I FTP'd up the unzipped J5.2.6Update pkg to the 4.4.14 site.  Getting Unknown column 'a.ordering' in 'order clause'.  Admin site getting 500 error.  Web research suggests that there are sql db update scripts that need to be run in the /administrator/components/com_admin/sql/updates/mysql folder.  I see 5.2.X scripts 6 of them.  Run them?  Or run all scripts newer than 4.4.14? There are 29 of those.

Thoughts?  Thanx.

**Reply by xillibit** [reply]
Yes you should apply all the 29 script from this directory

**Reply by AllonisDavid** [reply]
I ran the scripts in order (all 29) by editing each script in myPHPAdmin and adding the db prefix where needed to replace #_.  After that, the admin site loaded but with an error that required the plugin schemaorg to be disabled...then the admin site fully loaded.  I then attempted to update Kunena 6.4.9 but it's installer advises that "You don't have the minimum Joomla version requirement of J5.3.4"  So, I will attempt to update Joomla again.

**Reply by xillibit** [reply]
Kunena 6.4.9 can be installed on Joomla! 5.0.3, 5.14, 5.2.6, 5.3.4, 5.4.2 and 6.0.2

I have tested myself, the requirements aren't correct : 
<img width="3424" height="719" alt="image" src="https://github.com/user-attachments/assets/ec8c2163-71a8-45a7-a453-bf33282963b3" />

**Reply by AllonisDavid** [reply]
Do you expect to change the requirements and upload that?  Or is there a file that can be edited for 5.2.6 compatibility?  I am modding the files to 644 and directories to 755 currently as the Joomla updater complained.

**Reply by xillibit** [reply]
You can do the following change : https://github.com/Kunena/Kunena-Forum/commit/4d6e194a4f9905cdd11370ff6eeee2a31a914e2a in the package of K6.4.9 installation

**Reply by AllonisDavid** [reply]
made that file change, it then installed and now Call to a member function isClient() on null after installing Kunena
JROOT/plugins/system/kunena/kunena.php:122

Disabled Kunena and Admin now loads again.

**Reply by AllonisDavid** [reply]
seems the double period in the Joomla version script is a typo?
protected $minimumJoomla = '5..0.3

**Reply by AllonisDavid** [reply]
PHP message: PHP Deprecated:  Creation of dynamic property Kunena\\Forum\\Libraries\\Config\\KunenaConfig::$social is deprecated in /var/www/html/libraries/kunena/src/Config/KunenaConfig.php on line 1853; PHP message: PHP Deprecated:  Creation of dynamic property Kunena\\Forum\\Libraries\\Config\\KunenaConfig::$moderator_id is deprecated in /var/www/html/libraries/kunena/src/Config/KunenaConfig.php on line 1853; PHP message: PHP Deprecated:  Creation of dynamic property Kunena\\Forum\\Libraries\\Config\\KunenaConfig::$pickup_category is deprecated in /var/www/html/libraries/kunena/src/Config/KunenaConfig.php on line 1853; PHP message: PHP Deprecated:  Creation of dynamic property Kunena\\Forum\\Libraries\\Config\\KunenaConfig::$twitterConsumerKey is deprecated in /var/www/html/libraries/kunena/src/Config/KunenaConfig.php on line 1853; PHP message: PHP Deprecated:  Creation of dynamic property Kunena\\Forum\\Libraries\\Config\\KunenaConfig::$twitterConsumerSecret is deprecated in /var/www/html/libraries/kunena/src/Config/KunenaConfig.php on line 1853'

In the Forum run time see this:
1054 Unknown column 'socials' in 'field list'

**Reply by xillibit** [reply]
> seems the double period in the Joomla version script is a typo?
> protected $minimumJoomla = '5..0.3

Yes, it's a typo i had just fixed it : https://github.com/Kunena/Kunena-Forum/commit/0136bcac52c06e114d1f0bfec388cd9a4f592474

> In the Forum run time see this:
> 1054 Unknown column 'socials' in 'field list'

Did-you have opened the Kunena dashboard after to have installed Kunena 6.4.9 ? If yes, is-that showed messages which indicated that some updates had been done ?

**Reply by AllonisDavid** [reply]
Yes, Kunena dashboard was opened after installing 6.4.9.  I even deleted the config settings out of the database, and resaved them in the Kunena Config as I saw a forum post suggesting that helps with later PHP settings....but no subsequent change in the "social" error message. 1054 Unknown column 'socials' in 'field list' which you can see when you hover over the open browser tab.  We get a 404 no component found error on the page but only when logged into Joomla / Community Builder and attempting to access Kunena.  When logged out, Kunena opens fine.

**Reply by xillibit** [reply]
For the issue with socials it will be possible to fix in Kunena 7.0.2, i mean i have added a new button under tools which will that : 

[https://www.kunena.org/forum/kunena-7-0-0-support/169073-getting-404-after-latest-update-to-6-9-4-or-7-0-1?start=18#234598](https://www.kunena.org/forum/kunena-7-0-0-support/169073-getting-404-after-latest-update-to-6-9-4-or-7-0-1?start=18#234598)

**Reply by AllonisDavid** [reply]
Updated to K 7.0.0.2 and clicked the fix socials button on tools...An error has occurred.
1054 Unknown column 'x_social' in 'field list'

**Comment by AllonisDavid** [comment]
Hi,
When a user attempts to do a Quick Reply using latest 7.0.2 release, they
get a 404 error.  Here are log results.  Does that help with your
diagnostics?
https://allonis.com
Thank you!!!!
David Li


12 {main}
2026-03-06T13:01:11+00:00 CRITICAL 212.39.101.112 error Uncaught Throwable
of type Error thrown with message "Call to undefined method
Joomla\CMS\Document\HtmlDocument::addItem()". Stack trace: #0
[ROOT]/components/com_kunena/src/Controller/Application/Category/Feed/CategoryDisplay.php(99):
Kunena\Forum\Site\Controller\Application\Category\Feed\CategoryDisplay->createItem()
#1 [ROOT]/libraries/kunena/src/Controller/KunenaControllerDisplay.php(150):
Kunena\Forum\Site\Controller\Application\Category\Feed\CategoryDisplay->before()
#2 [ROOT]/components/com_kunena/src/Dispatcher/Dispatcher.php(107):
Kunena\Forum\Libraries\Controller\KunenaControllerDisplay->execute()
#3 [ROOT]/libraries/src/Component/ComponentHelper.php(361):
Kunena\Forum\Site\Dispatcher\Dispatcher->dispatch()
#4 [ROOT]/libraries/src/Application/SiteApplication.php(217):
Joomla\CMS\Component\ComponentHelper::renderComponent()
#5 [ROOT]/libraries/src/Application/SiteApplication.php(271):
Joomla\CMS\Application\SiteApplication->dispatch()
#6 [ROOT]/libraries/src/Application/CMSApplication.php(320):
Joomla\CMS\Application\SiteApplication->doExecute()
#7 [ROOT]/includes/app.php(58):
Joomla\CMS\Application\CMSApplication->execute()
#8 [ROOT]/index.php(51): require_once('...')
#9 {main}
2026-03-06T13:02:05+00:00 CRITICAL 185.191.171.17 error Uncaught Throwable
of type Twig\Error\RuntimeError thrown with message "An exception has been
thrown during the rendering of a template ("DOM Error 76: Unexpected end
tag : head
 while parsing:
<!doctype html>
<html><head>    <link rel="stylesheet"
href="gantry-engine://css-compiled/nucleus.css" type="text/css"/>
                <link rel="stylesheet" href="epsilon.scss"
type="text/css"/>            <br />
<b>Deprecated</b>:  trim(): Passing null to parameter #1 ($string) of type
string is deprecated in
<b>[ROOT]/libraries/gantry5/compat/vendor/twig/twig/src/Extension/CoreExtension.php</b>
on line <b>996</b><br />

                                     <link rel="stylesheet"
href="/media/gantry5/engines/nucleus/css-compiled/bootstrap5.css?69a3a543"
type="text/css" />
                        <link rel="stylesheet"
href="/media/system/css/joomla-fontawesome.min.css" type="text/css" />

                                    <link rel="stylesheet"
href="/media/plg_system_debug/css/debug.css" type="text/css"/>
